overrides/runner.rb

```diff
@@ -23,76 +23,122 @@ module Overrides
       def build(api, overrides, res_override_class = Overrides::ResourceOverride,
                 prop_override_class = Overrides::PropertyOverride)
+        override_info = OverrideInfo.new(overrides, prop_override_class, res_override_class)
         validator = Overrides::Validator.new(api, overrides)
         validator.run
-        build_product(api, overrides, resource: res_override_class, property: prop_override_class)
+        build_product(api, override_info)
       end
 
+      # Takes in a single property, single override, and a property override class and returns
+      # a brand new property with overrides applied.
+      # This is used exclusively for Ansible filters, which are regular properties that live
+      # outside of a Api::Product
       def build_single_property(api_property, property_override, prop_override_class)
-        build_property(api_property, property_override, { property: prop_override_class }, '')
+        overrides = SinglePropertyOverride.new(property_override, prop_override_class)
+        build_property(api_property, '', overrides, '')
       end
 
       private
 
       # Given a old Api::Product, and Overrides::ResourceOverrides,
       # returns a new Api::Product with overrides applied
-      def build_product(old_prod, all_overrides, override_classes)
+      def build_product(old_prod, overrides)
         prod = Api::Product.new
         old_prod.instance_variables
                 .reject { |o| o == :@objects }.each do |var_name|
-          if (all_overrides['product'] || {})[var_name]
-            prod.instance_variable_set(var_name, all_overrides['product'][var_name])
+          if overrides.product_overrides[var_name]
+            prod.instance_variable_set(var_name, overrides.product_overrides[var_name])
           else
             prod.instance_variable_set(var_name, old_prod.instance_variable_get(var_name))
           end
         end
         prod.instance_variable_set('@objects',
                                    old_prod.objects
                                            .map do |o|
-                                     build_resource(o, all_overrides[o.name],
-                                                    override_classes)
+                                     build_resource(o, overrides)
                                    end)
         prod
       end
 
       # Given a Api::Resource and Provider::Override::ResourceOverride,
       # return a new Api::Resource with overrides applied.
-      def build_resource(old_resource, res_override, override_classes)
-        res_override = override_classes[:resource].new if res_override.nil? || res_override.empty?
+      def build_resource(old_resource, overrides)
+        # Set up the ResourceOverride object.
+        res_override = overrides.resource_overrides(old_resource.name)
         res_override.validate
         res_override.apply old_resource
 
+        # Create new Api::Resource + apply all provider-specific values.
```

Comment:
This comment makes me think that `set_additional_values` wants to be named a little differently potentially? Assuming both call sites are doing the same thing.